mlflow/exceptions.py

```diff
@@ -119,7 +122,16 @@ def __init__(self, json):
             error_code,
             json["message"] if "message" in json else "Response: " + str(json),
         )
-        super().__init__(message, error_code=ErrorCode.Value(error_code))
+        try:
+            super().__init__(message, error_code=ErrorCode.Value(error_code))
+        except ValueError:
+            _logger.warning(
+                f"Received error code not recognized by MLflow: {error_code}, this may indicate "
+                "your request hits error outside the MLflow server, e.g., proxy server, IP blocker "
+                "or so."
+            )
+            super().__init__(message)
```

Comment:
For Databricks 403s, we *can* recognize the error code by mapping "permission_denied" to 403 or vice versa, right? Can we do that here?

Reply:
I don't know if we want to handle `403` specially - it's possible that other error codes are returned such as `404` or `429`. So I feel it's cleaner to keep the original message if no enum is mapped.

An alternative is to create a reverse map to [ERROR_CODE_TO_HTTP_STATUS](https://github.com/mlflow/mlflow/blob/5d13d6ec620a02de9a5e31201bf1becdb9722ea5/mlflow/exceptions.py#L28), which i feel is not quite necessary.